src/server/game/Maps/Map.cpp

```diff
@@ -516,6 +526,21 @@ bool Map::EnsureGridLoaded(const Cell &cell)
     return false;
 }
 
+void Map::GridMarkNoUnload(uint32 x, uint32 y)
+{
+    _gridNoUnload.insert(GridCoord(x, y).GetId());
```

Comment:
Each grid already has a way to prevent unloading it - https://github.com/TrinityCore/TrinityCore/blob/7874bee7bfb70e0e039f91173cff212e9572de09/src/server/game/Grids/NGrid.h#L57